snuba/settings.py

```diff
@@ -81,8 +82,12 @@
     "bootstrap.servers": os.environ.get("DEFAULT_BROKERS", "localhost:9092"),
 }
 STORAGE_BROKER_CONFIG: Mapping[str, Mapping[str, Any]] = {}
+
+# DEPRECATED, please use KAFKA_TOPIC_MAP instead
 STORAGE_TOPICS: Mapping[str, Mapping[str, Any]] = {}
 
+KAFKA_TOPIC_MAP: Mapping[str, str] = {}
```

Comment:
If I get this correctly. There should be two groups of configuration parameters for a topic and two ways to provide such configuraiton. These two dimensions are orthogonal:
groups of configuration parameters:
- topic creation config: This includes settings that are used to initialize a topic like partitions number and message timestamp type
- topic connection config: basically just the name. Which is also included in the creation config, so maybe this is not really necessary

ways to provide the configuration:
- application config. These parameters are not customizable and provided by the app: like message timestamp type
- custom config. This can be customized. Like partitions and topic name.

I think we should have a structure that allows capture these dimensions. 
Ignoring for a second connection parameters (as it seems only the name would be one of them), what about following the pattern we did for broker connection? There is a function that abstracts the config for a topic (https://github.com/getsentry/snuba/blob/9b59c2e46f7ba0fd5a673901b39c6c0987ba7f4a/snuba/utils/streams/backends/kafka.py#L654) that one loads the base parameters from settings and filters those we want to allow customizations onto, then applies the override). I think in this case we could apply first the standard application config and then the override parameters.

Since this configuration can be customized by the users, thus harder to refactor, I think it would be useful to get it in a good shape instead of solving only the specific problem.

I think taking inspiration from this tool may also be interesting: https://github.com/segmentio/topicctl